test/unit/org/apache/cassandra/schema/SchemaComparisonBetweenSSTablesAndCQLTablesTest.java

```diff
@@ -0,0 +1,364 @@
+        TableMetadata cqlTableMetadata = TableMetadata.builder(encodedKeyspaceName, TABLE_4_NAME, TableId.fromUUID(UUID.randomUUID())).addPartitionKeyColumn("col_txt", UTF8Type.instance).addClusteringColumn("col_uuid", UUIDType.instance) // is clustering column but should be second
+                                                      .addClusteringColumn("col_int", ReversedType.getInstance(Int32Type.instance))   // is clustering column but should be first
+                                                      .addRegularColumn("col_bool", BooleanType.instance).addRegularColumn("col_dec", DecimalType.instance).build();
+        assertThatExceptionOfType(org.apache.cassandra.exceptions.ConfigurationException.class).isThrownBy(() -> ssTableMetadata.validateTableStructureCompatibility(cqlTableMetadata));
+    }
+
+    /**
+     * Different column name for clustering column passes validation
+     * This is the expected behaviour as the clustering column names are not contained in the sstable metadata
+     */
+    @Test
+    public void testClusteringColumnNameMismatchPassesValidation()
+    {
+        TableMetadata ssTableMetadata = ssTableMetadataForTableMap.get(TABLE_2_NAME);
+        TableMetadata cqlTableMetadata = TableMetadata.builder(encodedKeyspaceName, TABLE_2_NAME, TableId.fromUUID(UUID.randomUUID())).addPartitionKeyColumn("col_txt", UTF8Type.instance).addRegularColumn("col_int", Int32Type.instance).addClusteringColumn("wrong_name", UUIDType.instance)   // is clustering column but wrong name
+                                                      .addRegularColumn("col_bool", BooleanType.instance).addRegularColumn("col_dec", DecimalType.instance).build();
+        assertThatCode(() -> ssTableMetadata.validateTableStructureCompatibility(cqlTableMetadata)).doesNotThrowAnyException();
+    }
+
+    // *** REGULAR COLUMN VALIDATION ADDITIONAL TESTS *** //
+
+    @Test
+    public void testRegularColumnMismatchFailsValidation_WrongColumnType()
+    {
+        TableMetadata ssTableMetadata = ssTableMetadataForTableMap.get(TABLE_1_NAME);
+        TableMetadata cqlTableMetadata = TableMetadata.builder(encodedKeyspaceName, TABLE_1_NAME, TableId.fromUUID(UUID.randomUUID())).addPartitionKeyColumn("col_txt", UTF8Type.instance).addRegularColumn("col_int", Int32Type.instance).addRegularColumn("col_uuid", UTF8Type.instance)  // is regular column but wrong type
+                                                      .addRegularColumn("col_bool", BooleanType.instance).addRegularColumn("col_dec", DecimalType.instance).build();
+        assertThatExceptionOfType(org.apache.cassandra.exceptions.ConfigurationException.class).isThrownBy(() -> ssTableMetadata.validateTableStructureCompatibility(cqlTableMetadata));
+    }
+
+    @Test
+    public void testRegularColumnNameMismatchPassesValidation()
+    {
+        // TODO this may have to be expected to pass validation legitimately
```

Comment:
I think the reason why this passes is that the validation only goes through the columns belonging to `this` table metadata and compares them with the columns, with the same names, present in the provided table metadata. In this case it won't find the column, as the provided table metadata has the same column but with a different name, and skips the validation.

Reply:
Yes, it makes sense. Is it something that should be fixed, or is it an acceptable and expected behaviour?

Reply:
Given that it was the original logic, I would guess it's expected.